### ISSUE TITLE
Add Tensorboard Integration

### ISSUE THREAD

**Issue by ryanleary** [ryanleary/patter#2]
Add tensorboard integration to log the following:
- [x] loss every `n` minibatches
- [x] every epoch:
  - [x] training loss
  - [x] dev loss/wer/cer
  - [ ] potentially wer/cer with a beam decoder/LM
- [ ] learning rate schedule
- [x] model weights (histograms)
- [x] image representation of first cnn weights
